### PR TITLE
OSIDB-2361: put advanced search criteria into query parameters

### PR DESCRIPTION
**Pull request by tboyetteredhat**
# [OSIDB-2361] [Advanced search: put search criteria into query parameters]

## Checklist:

- [x] Linting passed
- [x] Type checks passed
- [x] Tests suite passed
- [x] Commits consolidated
- [x] Changelog updated
- [x] Test cases added/updated
- [x] Jira ticket updated

## Summary:

[Replace with a brief summary of the changes introduced by this PR.]

## Changes:

[Replace with a detailed description of the changes made in this PR, including any new features, enhancements, bug fixes, or refactorings.]

## Considerations:

[Replace with any additional considerations, notes, or instructions for reviewers.]

### CODE REVIEW

**Review by C-Valen — APPROVED** [review]
LGTM

Just noting a small detail, there is a warn:
`Missing required prop: "query" 
  at <FlawSearchView class="osim-page-view" onVnodeUnmounted=fn<onVnodeUnmounted> ref=Ref< undefined > > `
  
I think `query` parameter could be set to optional on `FlawSearchView` as in some cases it's accessed without a query (like accessing advanced search in first instance)

**Comment by tboyetteredhat** [comment]
> LGTM
> 
> Just noting a small detail, there is a warn: `Missing required prop: "query" at <FlawSearchView class="osim-page-view" onVnodeUnmounted=fn<onVnodeUnmounted> ref=Ref< undefined > > `
> 
> I think `query` parameter could be set to optional on `FlawSearchView` as in some cases it's accessed without a query (like accessing advanced search for first instance)

Sure, I updated it.

**Review by C-Valen — APPROVED** [review]
Looks good!

File: src/components/__tests__/IssueSearchAdvanced.spec.ts [hunk 1]
```diff
@@ -38,16 +39,7 @@ describe('IssueSearchAdvanced', () => {
...
 
-  it('should emit on search button click', async () => {
-    const searchButton = wrapper.find('button[type="submit"]');
-    expect(searchButton.exists()).toBeTruthy();
-    await searchButton.trigger('submit');
-    await flushPromises();
-    const setFilterEvent = wrapper.emitted('set:filters');
-    expect(setFilterEvent[0][0]).toEqual({});
-  });
-
```
Comment (superbuggy) [inline src/components/__tests__/IssueSearchAdvanced.spec.ts:49]: do we want to keep this test?

File: src/utils/flawFields.ts [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  'cvss_scores',
+  'references',
+  'acknowledgments',
+  'embargoed',
+];
+
+export const flawFields = fieldsFor(ZodFlawSchema)
+  .filter((field) => includedFields.includes(field))
+  .flatMap((field) => fieldsMapping[field] || field)
+  .sort();
```
Comment (superbuggy) [inline src/utils/flawFields.ts:41]: let's put this in a new `constants/` dir in `src`.

File: src/components/__tests__/IssueSearchAdvanced.spec.ts [hunk 3]
```diff
@@ -57,7 +49,8 @@ describe('IssueSearchAdvanced', () => {
     expect(searchButton.exists()).toBeTruthy();
     await searchButton.trigger('submit');
     await flushPromises();
-    const setFilterEvent = wrapper.emitted('set:filters');
-    expect(setFilterEvent[0][0]).toEqual({ acknowledgments__name: 'test' });
+    expect(useRouter().replace).toHaveBeenCalled();
+    expect(useRouter().replace.mock.calls[0][0])
+      .toStrictEqual({ query: { query: 'search', acknowledgments__name: 'test' } });
```
Comment (superbuggy) [inline src/components/__tests__/IssueSearchAdvanced.spec.ts:54]: I think the router is not being tested here, but maybe I am missing something?

File: src/components/__tests__/IssueSearchAdvanced.spec.ts (hunk 1 above)
Comment (tboyetteredhat) [reply src/components/__tests__/IssueSearchAdvanced.spec.ts:49]: I think we need to remove this test

File: src/components/__tests__/IssueSearchAdvanced.spec.ts (hunk 3 above)
Comment (tboyetteredhat) [reply src/components/__tests__/IssueSearchAdvanced.spec.ts:54]: We don't need to test router. I think we just make sure route.replace is called with correct params.

**Review by superbuggy — APPROVED** [review]

**Review by C-Valen — APPROVED** [review]

**Pull request merged**